solidity_contracts/src/CairoPrecompiles/CairoLib.sol

```diff
@@ -156,4 +157,20 @@ library CairoLib {
         return libraryCall(classHash, functionSelector, data);
     }
 
+    /// @notice Performs a low-level call to send a message from the Kakarot to the Ethereum network.
+    /// @param toAddress The address of the Ethereum contract to send the message to.
+    /// @param data The data to send to the Ethereum contract.
+    function sendMessageToL1(
+        address toAddress,
+        uint248[] memory data
+    ) internal {
+        bytes memory callData = abi.encode(
```

Comment:
<img width="463" alt="image" src="https://github.com/kkrt-labs/kakarot/assets/18620296/d010cf23-2733-47a4-96ec-cb5e43a74494">
for the record, the calldata received at the precompile level be like, ie

- calldata[:32] => toAddress
- calldata[32:64] => data offset
- calldata[64:96] => data_len
- calldata[96:96 + data_len] => data